math-delimiters.el

```diff
@@ -85,12 +100,88 @@
           (const :tag "Dollars" ("$$" . "$$"))
           (cons :tag "Other" string string)))
 
+(defcustom math-delimiters-include-characters nil
+  "Characters to include in display math mode.
+If any of them are right behind an inline math equation, slurp
+them into the display math environment and barf them back out
+when converting to inline math."
+  :group 'math-delimiters
+  :type '(list character))
+
+(defcustom math-delimiters-compressed-display-math t
+  "Whether display math should be compressed.
+A value of `t' means that, when transforming from inline to
+display math, no additional line breaks are inserted.  For
+example, we go from
+
+    text-before \\(1 + 1\\) text-after
+
+to
+
+    text-before \\=\\[1 + 1\\] text-after.
+
+A value of `nil', however, would switch between the above inline
+math version and
+
+    text-before
+    \\=\\[
+      1 + 1
+    \\]
+    text-after."
+  :group 'math-delimiters
+  :type 'boolean)
+
+(defun math-delimiters--slurp-or-barf-characters (from-close)
+  "Slurp or barf characters.
+The list of values to consider is given by
+`math-delimiters-include-characters'.  Thus, include these
+characters in display math should they come right after an inline
+math statement, and exclude them in the other direction."
+  (cl-flet ((on-char-p (pos)
+                       (cl-loop for punct in math-delimiters-include-characters
+                                do (when (= punct pos)
+                                     (cl-return t)))))
```

Comment:
This is a big awkward; what I really want is to write this in one line with a function of the form `any: ∀a. Hom(a, Bool) × [a] → Bool` (to write it mathematically because we're talking TeX here :)) but I could not find anything like that outside of `dash.el`

Reply:
Well, here I'd simply use `(memq pos math-delimiters-include-characters)`, but there are functions like `any` in Emacs: `seq-find` and `cl-find-if`. (I don't think dash actually has much that isn't built-in, my impression is that people like it because they prefer the names it uses, not because Emacs doesn't have that functionality.)

Reply:
I changed this to use `cl-find` and made `math-delimiters-include-characters` a string instead of a list, since it feels easier to type `",.;"` than `'(?, ?. ?;)`.

Reply:
> Well, here I'd simply use (memq pos math-delimiters-include-characters), but there are functions like any in Emacs: seq-find and cl-find-if.

Oh that's really neat, thanks!  I suppose I should finally take a gander at the elisp manual after all